fresnel/geometry.py

```diff
@@ -183,6 +183,157 @@ def radius(self):
     def color(self):
         return util.array(self._geometry.getColorBuffer(), geom=self)
 
+
+class Box(Cylinder):
+    R""" Box geometry.
+
+    Generate a "box" shape.
+
+    Args:
+        scene (:py:class:`fresnel.Scene`): Add the geometry to this scene
+        box (`numpy.ndarray` or `array_like`): (``1x1, 1x3, or 1x6`` : ``float32``) Assumes 1x1 is cubic, 1x3 is orthorhombic, and 1x6 is triclinic.
+        radius (`numpy.ndarray` or `array_like`): (``N`` : ``float32``) - Radius of box edges.
+        color (`numpy.ndarray` or `array_like`): (``Nx2x3`` : ``float32``) - Color of box edges.
+
+    .. seealso::
+
+        :doc:`examples/01-Primitives/05-Box-geometry`
+            Tutorial: defining and setting box geometry properties
+        :doc:`examples/02-Advanced-topics/05-GSD-visualization`
+            Tutorial: Visualizing GSD files
+
+
+    Note:
+        The constructor arguments ``radius``, and ``color`` are optional. If you do not provide them,
+        they are initialized to 0.5 and black, respectively.
+
+    .. hint::
+        Avoid costly memory allocations and type conversions by specifying primitive properties in the appropriate
+        numpy array type.
+
+    Attributes:
+        points (:py:class:`fresnel.util.array`): Read or modify the box.
+        radius (:py:class:`fresnel.util.array`): Read or modify the radii of the box edges.
+        color (:py:class:`fresnel.util.array`): Read or modify the colors of the box.
+    """
+
+    def __init__(self,
+                 scene,
+                 box,
+                 radius=None,
+                 color=None,
+                 material=material.Material(solid=1.0, color=(0, 0, 0)),
+                 outline_material=material.Material(solid=1.0, color=(0, 0, 0)),
+                 outline_width=0.0):
+
+        self._geometry = scene.device.module.GeometryCylinder(scene._scene, 12)
+        self.material = material
+        self.outline_material = outline_material
+        self.outline_width = outline_width
+        self.box = box
+        self.points[:] = self._generate_points(self.box)
+
+        if radius is not None:
+            self.radius[:] = radius
+        else:
+            self.radius[:] = [0.5] * 12
+
+        if color is not None:
+            self.color[:] = color
+
+        self.scene = scene
+        self.scene.geometry.append(self)
+
+    def _generate_points(self, box):
+        '''
+        Helper function to take a box and calculate the 12 edges
```

Comment:
We have a somewhat more sophisticated form of this logic here: https://github.com/glotzerlab/plato/blob/9f6f0b6870a1dff5418d0cd0089f082d63501658/plato/draw/Box.py#L63-L96

I would consider adding a `@classmethod` like in plato and freud that allows users to pass generic box-like objects and will extract the appropriate info. Alternatively, just make the constructor accept generic box-like objects and duck-type them into the format you want.

Reply:
We decided to go the duck typing route to make updates to the box more straight forward.